build/builder.sh

```diff
@@ -59,6 +59,14 @@ fi
 # Absolute path to the toplevel cockroach directory.
 cockroach_toplevel="$(dirname $(cd $(dirname $0); pwd))"
 
+# We want our docker builds to use different go pkg/bin directories
+# than the host OS so that conflicting C++ toolchains do not cause
+# problems. See https://github.com/cockroachdb/cockroach/issues/7477.
+goos="linux"
```

Comment:
why do you need any of this? seems like we can use `docker_amd64{,_race}` below unconditionally.